### ISSUE TITLE
Is the error handling working?

### ISSUE THREAD

**Issue by mpgomez** [restx/restx#198]
Hi, 
I am trying to use the restx error handling, but when using it, after defining the exceptions and rules (I was refering to http://restx.io/docs/ref-errors-management.html):

```
throw RestxError.on(MsgSvcException.MsgSvcRules.ValidSubscriptionRef.class)
                    .set(MsgSvcException.MsgSvcRules.ValidSubscriptionRef.USER, ((MsgSvcSubscriptionDoesNotExist) ex).getUserName())
                    .set(MsgSvcException.MsgSvcRules.ValidSubscriptionRef.TOPIC, ((MsgSvcSubscriptionDoesNotExist) ex).getTopicName()).raise();
```

It cannot find the "on". I can't find any example of how to use error handling with rules in restx, or anything at all in google or the issues list.

**Comment by fcamblor** [comment]
(sorry for the long delay)

@nhumblot pinpointed this issue in [this restx website pull request](https://github.com/restx/restx.github.io/pull/14)

tldr; `RestxError.on()` doesn't exist (documentation were wrong) and you should rather inject a `RestxError` component and you will have the `on()` method on injected instance
